tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/wwwroot/app/dashboard/settings.controller.js

```diff
@@ -1265,7 +1284,67 @@
                     getTaxonomyHierarchy(data);
                 }
             }
-
+            vm.extraMatterFields = [];
+            function getAdditionalMatterProperties(data) {
+                vm.lazyloader = false;
+                var additionalMatterPropSettingName = configs.taxonomy.matterProvisionExtraPropertiesContentType;
+                if (data[additionalMatterPropSettingName] !== null && data[additionalMatterPropSettingName] !== undefined && data[additionalMatterPropSettingName] != "") {
+                    var optionsForGetmatterprovisionextraproperties = {
+                        Client: {
+                            Url: vm.clienturl
+                        },
+                        MatterExtraProperties: {
+                            ContentTypeName: data[additionalMatterPropSettingName]
+                        }
+                    }
+                    getmatterprovisionextraproperties(optionsForGetmatterprovisionextraproperties, function (result) {
+                        console.log(result);
+                        $("#divExtraMatterProps").removeClass("ng-hide");
+
+                        vm.extraMatterFields = result.Fields;
+                        angular.forEach(vm.extraMatterFields, function (field) {
+                            if (vm.additionalFieldValues !== null && vm.additionalFieldValues !== "") {
+                                angular.forEach(vm.additionalFieldValues, function (additionalfield) {
+                                    if (additionalfield.FieldName == field.fieldInternalName) {
+                                        if (additionalfield.IsDisplayInUI == null || additionalfield.IsDisplayInUI == undefined || additionalfield.IsDisplayInUI == '') {
+                                            additionalfield.IsDisplayInUI = 'false';
+                                        }
+                                        field.displayInUI = additionalfield.IsDisplayInUI;
+                                        if (additionalfield.IsMandatory == null || additionalfield.IsMandatory == undefined || additionalfield.IsMandatory == '') {
+                                            additionalfield.IsMandatory = 'false';
+                                        }
+                                        field.required = additionalfield.IsMandatory;
+                                    }
+                                })
+                            }
+                        });
+                        vm.lazyloader = true;
+                        console.log(vm.extraMatterFields);
+                    });
+                }
+                else {
+                    vm.lazyloader = true;
+                    $("#divExtraMatterProps").addClass("ng-hide");
+                    vm.extraMatterFields = [];
+                }
+            }
+
+            function getExtraMatterFieldValues() {
+                var Fields = [];
```

Comment:
This is the method to get the values after setting it after click on save button.